exercises/shared/.docs/tests.md

```diff
@@ -1,19 +1,154 @@
 # Tests
```

Comment:
There is a _lot_ of information in this file. As this file is distributed along with every downloaded exercise, we envisioned this to be a fairly small, to reduce the likelihood of people just not reading it. Quoting [the docs](https://exercism.org/docs/building/tracks/shared-files#h-file-tests-md):

> The instructions should be short and to the point.

What I often see done is to only mention the basics in this file, and then link to the track TESTS.md doc on the website for more information.

Reply:
I considered doing that.  The original was quite succinct and included a pointer to the website,  but I then got a lot of complaints about the pytest warnings and not being able to find information, and sort of went in the other direction. 

Let me see what I can do to pare this down a bit.

Reply:
Right. That's background information I didn't have, thanks!